### PR TITLE
Abstract Update Case logic into separate service

### PR DESCRIPTION
**Pull request by ClaudioCarmeli**
## Changes in this PR

- Case Update logic was removed from the controller action and placed in [a self-contained testable unit](https://github.com/DFE-Digital/buy-for-your-school/pull/530/files#diff-d7bdf92c88afa3f7c447a24240ca7aac6c7b56132e0e8d92fd28b80b047adefdR5) (`Support::UpdateCase`)
- Redirect user to case page, after successful agent assignment
- [Added spec helper](https://github.com/DFE-Digital/buy-for-your-school/pull/530/files#diff-d35fdbf1d42c668f8f4c92df27b8d3f9151e36132faade568aff709c45044109R77) (following pattern already present elsewhere in the test suite)

### CODE REVIEW

**Comment by peterdavidhamilton** [comment]
The definition of done would include updating the browser coverage in `/spec/features/support/case_edit_spec.rb`.
It might be desirable therefore to also redirect to the updated case.

File: app/controllers/support/cases_controller.rb [hunk 1]
```diff
@@ -13,20 +14,19 @@ def edit
...
-      # :nocov:
-      @current_case.agent = Agent.find_by(id: params.dig(:support_case, :agent))
-      @current_case.state = "open"
-      @current_case.save!
-      # :nocov:
-
-      redirect_to support_cases_path
+      UpdateCase.new(current_case, agent).call
+
+      redirect_to support_case_path(anchor: "case-history")
```
Comment (AndrewCowle) [inline app/controllers/support/cases_controller.rb:19]: Are you not displaying any confirmation message advising user something was updated?

**Review by AndrewCowle — DISMISSED** [review]

**Review by montychristo — APPROVED** [review]
LGTM

File: app/controllers/support/cases_controller.rb (hunk 1 above)
Comment (ClaudioCarmeli) [reply app/controllers/support/cases_controller.rb:19]: No, but that's a good idea. It's not in the A/C's so I thought we May want to add it in a separate PR / Jira ticket. Happy to add if it others are happy with that 👍

**Review by EdwinKruglov — APPROVED** [review]

File: spec/features/support/case_edit_spec.rb [hunk 2]
```diff
@@ -22,4 +22,12 @@
   it "displays the submit button" do
     expect(find_button("Assign")).to be_present
   end
+
+  context "when assigning agent to case" do
+    before { find_button("Assign").click }
+
+    it "is redirected to the case in question" do
```
Comment (peterdavidhamilton) [inline spec/features/support/case_edit_spec.rb:29]: Is this something that is seen in the browser? If so, can we get a minimal smoke test to assert that the user sees the change?

Comment (ClaudioCarmeli) [reply spec/features/support/case_edit_spec.rb:32]: Done 🎉  https://github.com/DFE-Digital/buy-for-your-school/pull/530/commits/ac3a23e0dffe10491bd94f0c44efe8a93f482539

**Review by AndrewCowle — APPROVED** [review]

File: spec/features/support/case_edit_spec.rb [hunk 3]
```diff
@@ -24,10 +24,25 @@
   end
 
   context "when assigning agent to case" do
-    before { find_button("Assign").click }
+    before do
+      find("label.govuk-radios__label", text: "John Lennon").click
+      find_button("Assign").click
+    end
 
     it "is redirected to the case in question" do
```
Comment (peterdavidhamilton) [inline spec/features/support/case_edit_spec.rb:32]: Could do with using the same kind of language in our specs like `redirects to the case`

File: spec/features/support/case_edit_spec.rb [hunk 4]
```diff
@@ -24,10 +24,25 @@
...
       expect(page).to have_a_support_case_path
     end
+
+    context "when returning to the main page" do
```
Comment (peterdavidhamilton) [inline spec/features/support/case_edit_spec.rb:36]: `when on the case management dashboard`

**Review by peterdavidhamilton — APPROVED** [review]

**Pull request merged**